### PR TITLE
Big refactor.

### PR DESCRIPTION
**Pull request by sirmarcel**
Hello, as discussed here a MONSTER pull request.

- Re-structured the package to be more in line with python standards, so imports should work as expected from outside the package itself.
- Rely on `schnetkit` for loading `schnetpack` models, and to supply a calculator for end-to-end tests.

The big caveat is that many tests are no longer passing, but it seems to be *only* because there is a hardcoded size of 61 for some neighborlist elements, which is now 60 for some reason. I don't know why, and I don't know enough to know where the 61 is coming from. We should definitely resolve this.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
+name = "schnax"
+version = "0.1.0"
+description = "schnet in jax"
+authors = ["Fabian Nagel <fabian.2.nagel@uni-konstanz.de>", "Marcel Langer <dev@marcel.science>"]
+
+[tool.poetry.dependencies]
```
Comment (fabiannagel) [inline pyproject.toml:7]: do we really want go all-in on bleeding edge dependencies from master branches? I'm almost sure that will mysteriously break at some point 😅

File: schnax/utils/schnetkit.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        weight_key='output_modules.0.out_net.1.out_net.0.weight',
+        bias_key='output_modules.0.out_net.1.out_net.0.bias',
+    )
+    set_params(
+        layer_key='SchNet/~/atomwise/~/linear_1',
+        weight_key='output_modules.0.out_net.1.out_net.1.weight',
+        bias_key='output_modules.0.out_net.1.out_net.1.bias',
+    )
+
+    return params
```
Comment (fabiannagel) [inline schnax/utils/schnetkit.py:69]: Wouldn't we move this to [SchNetKit](https://github.com/sirmarcel/schnetkit) anyways?

**Review by fabiannagel — COMMENTED** [review]
PR looks fine. Let's get the tests working first before merging. I'll have a look.

File: schnax/utils/schnetkit.py (hunk 2 above)
Comment (sirmarcel) [reply schnax/utils/schnetkit.py:69]: Move what? I think the mapping of `schnetpack` weights to `haiku` params is a job for `schnax`, no? `schnetkit` will give you a nice `dict` with the architecture to expect.

File: pyproject.toml (hunk 1 above)
Comment (sirmarcel) [reply pyproject.toml:7]: This is mostly for convenience while co-developing the other tools. `ase` has continuous integration so shouldn't break on the fly.

Comment (sirmarcel) [reply pyproject.toml:7]: (I don't mind making `ase` a normal dependency)

**Pull request merged**